File: 1448 - Count Good Nodes in Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution
{
private:
    void goodNodesH(TreeNode * node, int v, int & c)
    {
        if (!node) return;

        if (node->val >= v) 
        {
            c += 1;
            v = node->val;
        }

        goodNodesH(node->left, v, c);
        goodNodesH(node->right, v, c);
    }
    
public:
    int goodNodes(TreeNode* root)
    { 
        int c{0};
        goodNodesH(root, int(-10e4), c);
        return c;
    }
};

File: 1448 - Count Good Nodes in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        
        def goodNodesH(node, value):
            c = 0
            
            if node is None:
                return c
            
            if node.val >= value:
                c += 1
            
            value = max(value, node.val)
            
            c += goodNodesH(node.left, value)
            c += goodNodesH(node.right, value)
            
            return c
        
        return goodNodesH(root, root.val)